miniwin/miniwin/src/miniwin_ddraw.cpp

```diff
@@ -0,0 +1,188 @@
+#include "miniwin_ddraw.h"
+#include "miniwin_d3d.h"
+
+#include <SDL3/SDL.h>
+
+struct DirectDrawImpl : public IDirectDraw2, public IDirect3D2 {
+    DirectDrawImpl()
+    : m_refCount(1) {
+    }
+    // IUnknown interface
+	ULONG AddRef() override {
+        m_refCount += 1;
+        return m_refCount;
+    };
+	ULONG Release() override {
+        m_refCount -= 1;
+        if (m_refCount == 0) {
+            delete this;
+        }
+        return m_refCount;
+    }
+	HRESULT QueryInterface(const GUID& riid, void** ppvObject) override {
+        if (SDL_memcmp(&riid, &IID_IDirectDraw2, sizeof(GUID)) == 0) {
+            AddRef();
+            *ppvObject = dynamic_cast<IDirectDraw2*>(this);
+            return S_OK;
+        }
+        if (SDL_memcmp(&riid, &IID_IDirect3D2, sizeof(GUID)) == 0) {
+            AddRef();
+            *ppvObject = dynamic_cast<IDirect3D2*>(this);
```

Comment:
This is known to be of `IDirectDraw2` here, so no need for a dynamic cast, using static_cast faster, and RTTI can be an issue when targeting some platforms.

```suggestion
            *ppvObject = static_cast<IDirectDraw2*>(this);
            return S_OK;
        }
        if (SDL_memcmp(&riid, &IID_IDirect3D2, sizeof(GUID)) == 0) {
            AddRef();
            *ppvObject = static_cast<IDirect3D2*>(this);
```